### DISCUSSION TITLE
Whitelist Domain on Dashboard

### DISCUSSION THREAD

**Discussion by sebastian-baier** [Web3Auth/Web3Auth#1143] [Help]
Hello,
I tried to whitelist my domain on the dashboard but still got an error message in my webapp.
```
OpenLogin - RPC Error: could not validate redirect, please whitelist your domain: https://gymwallet.co for provided clientId ... at https://dashboard.web3auth.io.
    Also, this project is on mainnet network. Please ensure the the used Client ID belongs to this network. 
```
![image](https://user-images.githubusercontent.com/21202164/209918263-ee35a044-9123-440b-8ec1-ee872f034a44.png)

![image](https://user-images.githubusercontent.com/21202164/209918225-d4d952cc-a3f8-4d59-a437-76f42f900260.png)

my configuration looks like this:
```
const web3auth = new Web3Auth({
  web3AuthNetwork: 'testnet',
  clientId: clientId,
  chainConfig: {
    chainNamespace: CHAIN_NAMESPACES.EIP155,
    chainId: `0x${network.toString(16)}`,
    rpcTarget: url,
  },
  uiConfig: {
    appLogo: '/images/gym-wallet-white.svg',
    defaultLanguage: 'en',
    theme: 'dark',
    loginMethodsOrder: [
      'google',
      'apple',
      'facebook',
      'twitter',
      'github',
      'email_passwordless',
    ],
  },
  storageKey: 'local',
  authMode: 'DAPP',
});
```

I'm using infura RPC and want to try it out on testnet first.
Like this with localhost, all works fine, but not on my domain.
Help would be appreciated.

**Comment by Ni3l55** [comment]
I have the same issue. Also when using a Mainnet Web3 project

**Reply by Ni3l55** [reply]
It seems you fixed it on your site @sebastian-baier, could you please tell me how? Thanks

**Comment by thefallenmerc** [comment]
Bumping this up if anybody has the solution, facing the same problem.

**Comment by sebastian-baier** [comment]
I tried different things and it seems like the only network working is mainnet (for me)
so I switched every thing to mainnet and it works now, but don't know if that is supposed to be.
looks like this flag does nothing:
 ```web3AuthNetwork: 'testnet',```

**Comment by Ni3l55** [comment]
Also confirming it only worked for Mainnet for me. I was on Aqua before @thefallenmerc

**Comment by thefallenmerc** [comment]
Thanks @sebastian-baier @Ni3l55

That worked, testnet isnt working for some reason on urls other than localhost.

Also web3AuthNetwork: "mainnet" should be configured to make it work.

Now it is accessible from the whitelisted URLs

**Comment by thefallenmerc** [comment]
<img width="776" alt="image" src="https://user-images.githubusercontent.com/20546147/210242742-f092a163-0c66-4fe5-a37d-e6071fae5ecb.png">

**Comment by guru-web3** [comment]
please refer https://github.com/orgs/Web3Auth/discussions/1115

**Comment by byteZorvin** [comment]
Even I am facing this issue, do I really have to switch to mainnet to solve this?

**Reply by guru-web3** [reply]
no it is based on the network you chose while creating your solution on https://dashboard.web3auth.io/

**Reply by thefallenmerc** [reply]
@Mehul0011, if you have your app on testnet and trying to use it on anything other than localhost, you will face the white list issue even though you have whitelisted it in you testnet,
testnet whitelist won't work for some reason, so had to use mainnet.
I was able to overcome this by creating app on mainnet after testnet, whitelist on mainnet, use its clientId instead of testnet one.
This resolved issue for me, let me know if this helps.

**Reply by sebastian-baier** [reply]
@guru-web3 Not working for me and seems like for the others as well
For me nothing but mainnet is working with whitelisting

**Reply by byteZorvin** [reply]
Update: I was able to resolve the issue by specifying an extra paramter of web3AuthNetwork with value as testnet, which I previously did not have in the localhost code. With this its working for me even in testnet, Did not have to switch to mainnet.
![image](https://user-images.githubusercontent.com/75808688/211029999-9496364f-0680-4d72-a4e6-df6d449a348b.png)

**Reply by sebastian-baier** [reply]
strange that's what I did as well, if you look at the top
no idea what's the difference

**Reply by xpluscal** [reply]
just tried as well with testnet, seems to throw the same error. testing mainnet now.

**Comment by mykcryptodev** [comment]
I followed this discussion and the discussions linked by replies. I cannot get testnet working on anything other than local host (I am using Mumbai Polygon testnet)

1. Client ID is for a plug and play app that was created for testnet
2. web3AuthNetwork: “testnet” does not work

any advice?

**Comment by hanpanpan200** [comment]
```
const web3Auth = new Web3Auth({
    clientId,
    chainConfig: {
      chainNamespace: 'eip155',
      chainId: '0x13881', // hex of 80001, polygon testnet
      rpcTarget: MUMBAI_PROVIDER_URL,
      displayName: 'Polygon Testnet',
      blockExplorer: 'https://mumbai.polygonscan.com/',
      ticker: 'MATIC',
      tickerName: 'Matic',
    },
    uiConfig: {
      theme: 'light',
    },
    web3AuthNetwork: 'testnet', // Adding this line fixes the issue for me!!
  });
```
